hummingbot/market/bitfinex/bitfinex_api_order_book_data_source.py

```diff
@@ -0,0 +1,509 @@
+            raw_symbol = f"{price['base']}-{price['quote']}"
+            symbol = f"{price['base']}{price['quote']}"
+            prices.append(
+                {
+                    "symbol": symbol,
+                    "volume": price["volume"] * price["price"]
+                }
+            )
+            converters[price["base"]] = price["price"]
+            del raw_prices[raw_symbol]
+
+        for raw_symbol, item in raw_prices.items():
+            symbol = f"{item['base']}{item['quote']}"
+            if item["base"] in converters:
+                prices.append(
+                    {
+                        "symbol": symbol,
+                        "volume": item["volume"] * converters[item["base"]]
+                    }
+                )
+                if item["quote"] not in converters:
+                    converters[item["quote"]] = item["price"] / converters[item["base"]]
+                continue
+
+            if item["quote"] in converters:
+                prices.append(
+                    {
+                        "symbol": symbol,
+                        "volume": item["volume"] * item["price"] * converters[item["quote"]]
+                    }
+                )
+                if item["base"] not in converters:
+                    converters[item["base"]] = item["price"] * converters[item["quote"]]
+                continue
+
+            prices.append({"symbol": symbol, "volume": NaN})
+
+        return prices
+
+    def _get_tracked_order_by_id(self, order_id: int):
+        if order_id in self._tracked_book_entries:
+            return self._tracked_book_entries[order_id]
+        else:
+            return {"order": None, "side": None}
+
+    def _track_order(self, order_id: int, order: OrderBookRow, side: str):
+        self._tracked_book_entries[order_id] = {"order": order, "side": side}
+
+    def _untrack_order(self, order_id):
+        if order_id in self._tracked_book_entries:
+            del self._tracked_book_entries[order_id]
+
+    @classmethod
+    @async_ttl_cache(ttl=REQUEST_TTL, maxsize=CACHE_SIZE)
+    async def get_active_exchange_markets(cls) -> pd.DataFrame:
+        async with aiohttp.ClientSession() as client:
+            async with client.get(f"{BITFINEX_REST_URL}/tickers?symbols=ALL") as tickers_response:
+                tickers_response: aiohttp.ClientResponse = tickers_response
+
+                status = tickers_response.status
+                if status != RESPONSE_SUCCESS:
+                    raise IOError(
+                        f"Error fetching active Bitfinex markets. HTTP status is {status}.")
+
+                data = await tickers_response.json()
+
+                raw_prices = cls._get_prices(data)
+                prices = cls._convert_volume(raw_prices)
+
+                all_markets: pd.DataFrame = pd.DataFrame.from_records(data=prices, index="symbol")
+
+                return all_markets.sort_values("volume", ascending=False)
```

Comment:
`get_active_exchange_markets` require the following columns, `USDVolume`, `baseAsset`, `quoteAsset`

Reply:
Hi @CrimsonJacket ! How USDVolume is expected to be calculated? Is this per 24-hour period? If yes, should it be re-calculated using which quote currency price in USD? I am not sure we know quote price in USD in general case.

Reply:
Bitfinex does not give USDVolume datapoint, we had to write code to calculate it.